### PR TITLE
Add new release workflows

### PR DESCRIPTION
**Pull request by sethvargo**
This will also require updates to minty's config.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+on:
+  push:
+    branches:
+      - 'main'
+      - 'release/**/*'
+
+jobs:
+  release:
+    runs-on: 'ubuntu-latest'
+    if: '${{ startsWith(github.event.head_commit.message, ''Release: v'') && github.event.head_commit.author.email == vars.TOKEN_MINTER_GITHUB_EMAIL }}'
```
Comment (verbanicm) [inline .github/workflows/release.yml:26]: I prefer the following syntax for readability:

```suggestion
    if: |-
      startsWith(github.event.head_commit.message, 'Release: v') &&
      github.event.head_commit.author.email == vars.TOKEN_MINTER_GITHUB_EMAIL
```

File: .github/actions/create-release/action.yml [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+          // entrypoint.
+          const matches = context.payload.head_commit.message.match(/Release: v(?<version>[^\ ]+)/i);
+          if (!matches || !matches.groups) {
+            core.setFailed(`Commit message does not contain a version`)
+            return;
+          }
+          let releaseVersion = matches.groups.version;
+          while(releaseVersion.charAt(0).toLowerCase() === 'v') {
+            releaseVersion = releaseVersion.substr(1);
+          }
```
Comment (verbanicm) [inline .github/actions/create-release/action.yml:113]: Should this be an `if`?

File: .github/actions/create-release/action.yml [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+          // Ensure the commit is signed.
+          const commitResult = await github.rest.repos.getCommit({
+            owner: context.repo.owner,
+            repo: context.repo.repo,
+            ref: context.payload.head_commit.id,
+          })
+          if (!commitResult.data.commit.verification.verified) {
+            core.setFailed(`Commit is not signed`)
+            return;
+          }
```
Comment (verbanicm) [inline .github/actions/create-release/action.yml:100]: Do we need this? I feel like the ruleset should enforce this?

File: .github/actions/create-release/action.yml (hunk 2 above)
Comment (sethvargo) [reply .github/actions/create-release/action.yml:113]: No, this handles `vvvvv1.2` -> `1.2`

File: .github/actions/create-release/action.yml (hunk 3 above)
Comment (sethvargo) [reply .github/actions/create-release/action.yml:100]: Probably not? When this was a reusable workflow, this was ensuring that the email address wasn't forged.

Comment (sethvargo) [reply .github/actions/create-release/action.yml:100]: Actually, we do need this. It's possible to sign a commit as a bot (which will pass rulesets), but fail validation. This isn't just checking that the commit is signed, but it's also checking that the commit is _verified_; rulesets don't do that as far as I can tell.

**Review by verbanicm — APPROVED** [review]

**Review by verbanicm — APPROVED** [review]

**Pull request merged**